packages/checkout-storefront/src/sections/BillingAddressSection/BillingAddressSection.tsx

```diff
@@ -57,11 +59,13 @@ export const BillingAddressSection = () => {
 
   const updateBillingAddress = useCallback(
     async ({ autoSave, ...addressInput }: AddressFormData) => {
-      const result = await checkoutBillingAddressUpdate({
-        checkoutId,
-        billingAddress: getAddressInputData(addressInput),
-        validationRules: getAddressVlidationRulesVariables(autoSave),
-      });
+      const result = await checkoutBillingAddressUpdate(
+        getQueryVarsWithLocale({
```

Comment:
I think it is the wrong abstraction for this. `checkoutBillingAddressUpdate` doesn't care about query variables. These values could be coming from anywhere else, not necessarily query variables.